rules/aip0123/resource_type_name.go

```diff
@@ -30,14 +33,39 @@ var resourceTypeName = &lint.MessageRule{
 	},
 	LintMessage: func(m *desc.MessageDescriptor) []lint.Problem {
 		resource := utils.GetResource(m)
-		if strings.Count(resource.GetType(), "/") != 1 {
+		resourceType := resource.GetType()
+		if strings.Count(resourceType, "/") != 1 {
 			return []lint.Problem{{
 				Message:    "Resource type names must be of the form {Service Name}/{Type}.",
 				Descriptor: m,
 				Location:   locations.MessageResource(m),
 			}}
 		}
-
+		typeName := strings.Split(resourceType, "/")[1]
+		if !unicode.IsUpper(rune(typeName[0])) {
+			return []lint.Problem{{
+				Message:    fmt.Sprintf("Type %q must be UpperCamelCase", typeName),
+				Descriptor: m,
+				Location:   locations.MessageResource(m),
+			}}
+		}
+		matched, err := regexp.MatchString("^[\\p{L}A-Za-z0-9]+$", typeName)
```

Comment:
Can you move this regexp into a package-global, unexported `var alphaNumericTypeName` and use [`regexp.MustCompile`](https://pkg.go.dev/regexp#MustCompile) instead? The `err` here is for compiling the regexp and I think things should fail much earlier if this regexp is wrong - most other regexps in this project are compiled via `MustCompile`. This way, we don't need to handle the `err`/report it to users on their proto.

Reply:
great! that's much better. Will do.